### PR TITLE
Refactored 3sum to decrease CC. Also changed tests accordingly

### LINKED ISSUE #58: Refactor three_sum.py to reduce CC
Title says it all :)
Currently has a CC of 11. Need to get it to 8 or lower.

### PR DESCRIPTION
**Pull request by ArnPellesGit**
Fixes #58

### CODE REVIEW

File: manual_evaluations/part2_coverage/refactored_functions/arrays/three_sum_improved.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+##
+
+##
+# This is a very naive implementation of the three_sum problem.
+# Given an array, it returns all unique triples that sum to 0.
+# It does so by bruteforcing - testing every single combination of triples
+# and checks if this triple sums to 0. If it does, add that triple to res
+# It then removes duplicates and returns the result.
+# It has a time complexity of n^3 where n = size of the array.
+# The time complexity is garbage but this reduces the CC from 11 to 4.
```
Comment (ArnPellesGit) [inline manual_evaluations/part2_coverage/refactored_functions/arrays/three_sum_improved.py:17]: freakin typo should be 5 not 4.

Comment (capitanu) [reply manual_evaluations/part2_coverage/refactored_functions/arrays/three_sum_improved.py:17]: Are you going to fix this tho? :D

**Review by capitanu — COMMENTED** [review]
Also probably add some files that prove the CC before and after

File: manual_evaluations/part2_coverage/refactored_functions/arrays/three_sum_improved.py (hunk 1 above)
Comment (ArnPellesGit) [reply manual_evaluations/part2_coverage/refactored_functions/arrays/three_sum_improved.py:17]: Hah not unless this wasn’t approved ^^ if it had request changes, I’d change it

**Comment by ArnPellesGit** [comment]
> Also probably add some files that prove the CC before and after

Figure that’s necessary? I find those lizard outputs so cluttery

**Comment by capitanu** [comment]
> > Also probably add some files that prove the CC before and after
> 
> Figure that’s necessary? I find those lizard outputs so cluttery

They are not the nicest, but when we will present, we can't really run lizard on the previous implementation unless we go in the root of the project, and we kind of all have that files, so it would be nice :D If you don't want to, I will approve this now and you can just merge it

File: manual_evaluations/part2_coverage/refactored_functions/arrays/three_sum_improved.py (hunk 1 above)
Comment (capitanu) [reply manual_evaluations/part2_coverage/refactored_functions/arrays/three_sum_improved.py:17]: I don't really get what you mean :D You don't need to get "changes requested" in order to fix something you figured was wrong

**Comment by ArnPellesGit** [comment]
> > > Also probably add some files that prove the CC before and after
> > 
> > 
> > Figure that’s necessary? I find those lizard outputs so cluttery
> 
> They are not the nicest, but when we will present, we can't really run lizard on the previous implementation unless we go in the root of the project, and we kind of all have that files, so it would be nice :D If you don't want to, I will approve this now and you can just merge it

Nah you're right, I'll do it rn and commit to this PR :)

File: manual_evaluations/part2_coverage/refactored_functions/arrays/three_sum_improved.py (hunk 1 above)
Comment (ArnPellesGit) [reply manual_evaluations/part2_coverage/refactored_functions/arrays/three_sum_improved.py:17]: Haha I meant that it was such a small typo that I wouldn't fix it unless someone else requested it :P

Comment (ArnPellesGit) [reply manual_evaluations/part2_coverage/refactored_functions/arrays/three_sum_improved.py]: This is now fixed

**Review by capitanu — APPROVED** [review]

**Pull request merged**